src/qt/receiverequestdialog.cpp

```diff
@@ -51,42 +45,57 @@ void ReceiveRequestDialog::setModel(WalletModel *_model)
 void ReceiveRequestDialog::setInfo(const SendCoinsRecipient &_info)
 {
     this->info = _info;
-    update();
-}
+    setWindowTitle(tr("Request payment to %1").arg(info.label.isEmpty() ? info.address : info.label));
+    QString uri = GUIUtil::formatBitcoinURI(info);
 
-void ReceiveRequestDialog::update()
-{
-    if(!model)
-        return;
-    QString target = info.label;
-    if(target.isEmpty())
-        target = info.address;
-    setWindowTitle(tr("Request payment to %1").arg(target));
+#ifdef USE_QRCODE
+    if (ui->qr_code->setQR(uri, info.address)) {
+        connect(ui->btnSaveAs, &QPushButton::clicked, ui->qr_code, &QRImageWidget::saveImage);
+    } else {
+        ui->btnSaveAs->setEnabled(false);
+    }
+#else
+    ui->btnSaveAs->hide();
+    ui->qr_code->hide();
+#endif
 
-    QString uri = GUIUtil::formatBitcoinURI(info);
-    ui->btnSaveAs->setEnabled(false);
-    QString html;
-    html += "<html><font face='verdana, arial, helvetica, sans-serif'>";
-    html += "<b>"+tr("Payment information")+"</b><br>";
-    html += "<b>"+tr("URI")+"</b>: ";
-    html += "<a href=\""+uri+"\">" + GUIUtil::HtmlEscape(uri) + "</a><br>";
-    html += "<b>"+tr("Address")+"</b>: " + GUIUtil::HtmlEscape(info.address) + "<br>";
-    if(info.amount)
-        html += "<b>"+tr("Amount")+"</b>: " + BitcoinUnits::formatHtmlWithUnit(model->getOptionsModel()->getDisplayUnit(), info.amount) + "<br>";
-    if(!info.label.isEmpty())
-        html += "<b>"+tr("Label")+"</b>: " + GUIUtil::HtmlEscape(info.label) + "<br>";
-    if(!info.message.isEmpty())
-        html += "<b>"+tr("Message")+"</b>: " + GUIUtil::HtmlEscape(info.message) + "<br>";
-    if(model->isMultiwallet()) {
-        html += "<b>"+tr("Wallet")+"</b>: " + GUIUtil::HtmlEscape(model->getWalletName()) + "<br>";
+    ui->uri_content->setText("<a href=\"" + uri + "\">" + GUIUtil::HtmlEscape(uri) + "</a>");
```

Comment:
In a label, this becomes clickable.

I'm not sure why we linkify it (so it looks like a link?), but previously it wasn't a clickable link

Reply:
> I'm not sure why we linkify it (so it looks like a link?)...

It seems so.

> ... but previously it wasn't a clickable link

Agree, this change should be reverted.

Reply:
> Agree, this change should be reverted.

Done in the latest push.